### DISCUSSION TITLE
Sender-address truncated after first plus-char

### DISCUSSION THREAD

**Discussion by chaosben** [rspamd/rspamd#3746] [Q&A] [answered]
Hi guys!

I'm wondering, why the local-part of the sender address handed over to rspamd (via milter) is truncated after the first "+" char. Is that the way it's supposed to be? Or is that adjustable?

Thanks for thinking about!

_Benjamin_

**Comment by fatalbanana** [comment, marked as answer]
There is a rule called `PLUS_ALIASES` which normalises sender addresses - the assumption is that 'plus addressing' is in play which mightn't necessarily be true.

The original addresses are retained but there is a dedicated API to get at them.

You could disable the rule in question or there may be other options depending on what you're aiming to solve.

**Reply by chaosben** [reply]
Thanks for your fast and good answer! 😊

Our "problem" is a smtp.from like `SRS0+139bc5624f5afb9f=J4=origdomain.ch=noreply@emailservice.ch`.
As of p. 3.1. at the [paper describing SRS](http://www.libsrs2.org/srs/srs.pdf) the first delimiter after SRS[01] can be any of =,+,-. 
The "emailservice" troubling us, uses the plus-char. Therefore the default-configured Rspamd truncates the localpart.

So my (hopefully last) question is: who is responsible for handling this kind of address correctly?

**Reply by moisseev** [reply]
> So my (hopefully last) question is: who is responsible for handling this kind of address correctly?

Since SRS is not a standard, it is solely at "emailservice's" own risk.

On the Rspamd side it is possible to add a prefilter for SRS addresses normalization, but email services use different address formats. 
For instance Office 365:
```
<Forwarding Mailbox Username>+SRS=<Hash>=<Timestamp>=<Original Sender Domain>=<Original Sender Username>@<Forwarding Mailbox Domain>
```

**Reply by chaosben** [reply]
OK, that helps us.
Thank you very much! 😍
